repository.go

```diff
@@ -0,0 +1,123 @@
+package function
+
+import (
+	"errors"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"github.com/markbates/pkger"
+)
+
+// Path to builtin repositories.
+// note: this constant must be defined in the same file in which it is used due
+// to pkger performing static analysis on source files separately.
+const builtinRepositories = "/templates"
+
+// Repository
+type Repository struct {
+	Name      string
+	Templates []Template
+	Runtimes  []string
+}
+
+// NewRepository from path.
+// Represents the file structure of 'path' at time of construction as
+// a Repository with Templates, each of which has a Name and its Runtime.
+// a convenience member of Runtimes is the unique, sorted list of all
+// runtimes
+func NewRepositoryFromPath(path string) (Repository, error) {
+	// TODO: read and use manifest if it exists
+
+	r := Repository{
+		Name:      filepath.Base(path),
+		Templates: []Template{},
+		Runtimes:  []string{}}
+
+	// Each subdirectory is a Runtime
```

Comment:
I suppose a `manifest.yaml` might specify either includes/excludes or have some other mechanism to make sure, we don't end up displaying `test` or something that's not actually a runtime dir?

Reply:
Exactly what I was thinking as well.  Directory structure a sensible default, `manifset.yaml` an optional way to override things (how many things and exactly _how_ ?  That's still a bit mind-murky)